modyn/supervisor/internal/grpc_handler.py

```diff
@@ -276,69 +278,47 @@ def start_training(
             optimizer_config["source"] = optimizer["source"]
             optimizer_config["param_groups"] = []
             for param_group in optimizer["param_groups"]:
-                config_dict = param_group["config"] if "config" in param_group else {}
+                config_dict = param_group.get("config") or {}
                 optimizer_config["param_groups"].append({"module": param_group["module"], "config": config_dict})
             optimizers_config[optimizer["name"]] = optimizer_config
 
         lr_scheduler_configs = {}
-        if "lr_scheduler" in pipeline_config["training"]:
+        if pipeline_config["training"].get("lr_scheduler"):
```

Comment:
why this change?

Reply:
Using the `in` operator for dicts isn't safe in python as it semantically doesn't always reflect what is intended. Dumping pydantic models can e.g. yield `None` values for dict fields. `in` treats the keys as present while no value is present.